templates/single.html

```diff
@@ -1,47 +1,57 @@
 <!-- wp:template-part {"slug":"header","tagName":"header"} /-->
 
 <!-- wp:group {"tagName":"main"} -->
-<main class="wp-block-group"><!-- wp:group {"layout":{"inherit":true}} -->
-<div class="wp-block-group"><!-- wp:post-title {"level":1,"align":"wide"} /-->
-
-<!-- wp:post-featured-image {"align":"wide"} /-->
-
-<!-- wp:separator {"align":"wide","className":"is-style-wide"} -->
-<hr class="wp-block-separator alignwide is-style-wide"/>
-<!-- /wp:separator --></div>
-<!-- /wp:group -->
-
-<!-- wp:spacer {"height":"var(--wp--preset--spacing--30)"} -->
-<div style="height:var(--wp--preset--spacing--30)" aria-hidden="true" class="wp-block-spacer"></div>
-<!-- /wp:spacer -->
-
-<!-- wp:post-content {"layout":{"inherit":true}} /-->
-
-<!-- wp:spacer {"height":"var(--wp--preset--spacing--30)"} -->
-<div style="height:var(--wp--preset--spacing--30)" aria-hidden="true" class="wp-block-spacer"></div>
-<!-- /wp:spacer -->
-
-<!-- wp:group {"layout":{"inherit":true}} -->
-<div class="wp-block-group"><!-- wp:group {"layout":{"type":"flex"}} -->
-<div class="wp-block-group"><!-- wp:post-date {"format":"F j, Y"} /-->
-
-<!-- wp:post-author {"showAvatar":false} /-->
-
-<!-- wp:post-terms {"term":"category"} /-->
-
-<!-- wp:post-terms {"term":"post_tag"} /--></div>
-<!-- /wp:group -->
-
-<!-- wp:spacer {"height":"var(--wp--preset--spacing--30)"} -->
-<div style="height:var(--wp--preset--spacing--30)" aria-hidden="true" class="wp-block-spacer"></div>
-<!-- /wp:spacer -->
-
-<!-- wp:separator {"className":"is-style-wide"} -->
-<hr class="wp-block-separator is-style-wide"/>
-<!-- /wp:separator -->
-
-<!-- wp:post-comments /--></div>
-<!-- /wp:group --></main>
+<main class="wp-block-group">
+	<!-- wp:group {"layout":{"inherit":true}} -->
+	<div class="wp-block-group">
+		<!-- wp:post-featured-image {"align":"wide"} /-->
+		<!-- wp:post-title {"level":1,"align":"wide","style":{"spacing":{"margin":{"top":"7rem","bottom":"7rem"}}}} /-->
+	</div>
+	<!-- /wp:group -->
+
+	<!-- wp:post-content {"layout":{"inherit":true}} /-->
+
+	<!-- wp:group {"style":{"spacing":{"margin":{"top":"7rem","bottom":"7rem"}}},"layout":{"inherit":true}} -->
+	<div class="wp-block-group" style="margin-top:7rem;margin-bottom:7rem">
+		<!-- wp:separator {"opacity":"css","align":"wide","className":"is-style-wide"} -->
+		<hr class="wp-block-separator alignwide has-css-opacity is-style-wide"/>
+		<!-- /wp:separator -->
+
+		<!-- wp:columns {"align":"wide","style":{"spacing":{"margin":{"top":"var(--wp--preset--spacing--70)"},"blockGap":"var(--wp--preset--spacing--70)"}},"fontSize":"small"} -->
```

Comment:
> "blockGap":"var(--wp--preset--spacing--70)"

Using the spacing preset inside blockGap does not seem to work.

Reply:
@andrewserong I am trying to use a spacing preset for a blockGap in a columns block, but in the block markup in the HTML file, not in theme.json. I can't get it to work. Is this known? I could not find an issue for it, https://github.com/WordPress/gutenberg/pull/43296 is the closest.
The gap CSS is missing from the columns .wp-container-id in the editor.
If I change it to a different value like 2rem, the CSS is added correctly.
(I am testing with Gutenberg trunk)

Reply:
Hi @carolinan! It isn't _quite_ working just yet, that PR unfortunately missed a couple of things to get it working properly. I'm currently working on a fix (and opt-in to the spacing controls) over in https://github.com/WordPress/gutenberg/pull/43466, which I hope to get in this week (it isn't yet ready for review, there's still a few things to tweak)

Reply:
Thank you :) Then I think it can be kept in the template.

Reply:
One thing to watch out for when using spacing presets (also for padding and margin) is that the editor will serialize the values to post content using the form `var:preset|spacing|30` instead of using the `var()` function directly, in the JSON part of the block markup. For example, this is how the Group block with padding and blockGap values currently looks in my local environment when added via the UI:

```html
<!-- wp:group {"layout":{"type":"constrained"},"style":{"spacing":{"padding":{"top":"var:preset|spacing|40","right":"var:preset|spacing|40","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40"},"blockGap":"var:preset|spacing|30"}}} -->
<div class="wp-block-group" style="padding-top:var(--wp--preset--spacing--40);padding-right:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--40)"><!-- wp:paragraph -->
<p>A paragraph in a group block</p>
<!-- /wp:paragraph --></div>
<!-- /wp:group -->
```

In that example, note that the JSON section uses the form `var:preset|spacing|40`, whereas the inline style uses the `var()` function in the HTML markup. Because blockGap is rendered on the server, it doesn't appear in the inline style.

Hope that helps!

Reply:
@carolinan just a heads-up that #43466 has now been merged, so the block spacing UI should now be working correctly with presets. Hopefully that'll make it a little easier building out the parts of the theme that depend on block spacing presets. I _think_ you'll probably wind up needing to update the markup to use the `var:preset|spacing|40` syntax in the block comments as the server rendering might not like the `var()` style, but I haven't manually tested it. Please let me know know (or add any issues) if you run into any issues with the UI changes as there were a few edge cases that needed to be handled in #43466 to get everything working correctly 🙂